src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -1,16 +1,879 @@
+            ); //Keep track of time left after all fixed loop are done
+
+            //UPDATING HYDRAULICS AT FIXED STEP
+            for _cur_loop in 0..num_of_update_loops {
+                //Base logic update based on overhead Could be done only once (before that loop) but if so delta time should be set accordingly
+                self.update_hyd_logic_inputs(&min_hyd_loop_timestep, &overhead_panel, &ct);
+
+                //Process brake logic (which circuit brakes) and send brake demands (how much)
+                self.hyd_brake_logic
+                    .update_brake_demands(&min_hyd_loop_timestep, &self.green_loop);
+                self.hyd_brake_logic.send_brake_demands(
+                    &mut self.braking_circuit_norm,
+                    &mut self.braking_circuit_altn,
+                );
+
+                //UPDATE HYDRAULICS FIXED TIME STEP
+                self.ptu.update(&self.green_loop, &self.yellow_loop);
+                self.engine_driven_pump_1.update(
+                    &min_hyd_loop_timestep,
+                    &self.green_loop,
+                    &engine1,
+                );
+                self.engine_driven_pump_2.update(
+                    &min_hyd_loop_timestep,
+                    &self.yellow_loop,
+                    &engine2,
+                );
+                self.yellow_electric_pump
+                    .update(&min_hyd_loop_timestep, &self.yellow_loop);
+                self.blue_electric_pump
+                    .update(&min_hyd_loop_timestep, &self.blue_loop);
+
+                self.rat.update(&min_hyd_loop_timestep, &self.blue_loop);
+                self.green_loop.update(
+                    &min_hyd_loop_timestep,
+                    Vec::new(),
+                    vec![&self.engine_driven_pump_1],
+                    Vec::new(),
+                    vec![&self.ptu],
+                );
+                self.yellow_loop.update(
+                    &min_hyd_loop_timestep,
+                    vec![&self.yellow_electric_pump],
+                    vec![&self.engine_driven_pump_2],
+                    Vec::new(),
+                    vec![&self.ptu],
+                );
+                self.blue_loop.update(
+                    &min_hyd_loop_timestep,
+                    vec![&self.blue_electric_pump],
+                    Vec::new(),
+                    vec![&self.rat],
+                    Vec::new(),
+                );
+
+                self.update_hyd_avail_states();
+
+                self.braking_circuit_norm
+                    .update(&min_hyd_loop_timestep, &self.green_loop);
+                self.braking_circuit_altn
+                    .update(&min_hyd_loop_timestep, &self.yellow_loop);
+                self.braking_circuit_norm.reset_accumulators();
+                self.braking_circuit_altn.reset_accumulators();
+            }
+
+            //UPDATING ACTUATOR PHYSICS AT "FIXED STEP / ACTUATORS_SIM_TIME_STEP_MULT"
```

Comment:
**suggestion**: Avoid code-like comments, instead split into functions with clear names describing what is happening.